Forgot password

OrigamiStarz opened the pull request:
## Tracking Info

Resolves #5 

## Changes

- Created and styled forgotPassword.tsx page
- Created and styled forgotPasswordForm and forgotPasswordEmailSent components
- Added optional props to TextBox and LoginButton to handle functionality
  - forgotPasswordEmailSent component displays the email typed by the user (currently commented out to see how designers want to handle long emails)
  - Entering inside TextBox does not reset the form/page

## Testing

- Added `src/app/pages/forgotPassword.tsx` as a component in `src/app/page.tsx` 

## Confirmation of Change

<img width="1440" alt="Screenshot 2025-01-25 at 4 52 29 PM" src="https://github.com/user-attachments/assets/933bfb74-a1af-4931-a93f-67161c346bda" />
<img width="1440" alt="Screenshot 2025-01-25 at 4 52 45 PM" src="https://github.com/user-attachments/assets/7ef92dc0-ce5d-4b49-8791-800ff8f9d40c" />
<img width="1440" alt="Screenshot 2025-01-25 at 5 20 48 PM" src="https://github.com/user-attachments/assets/f6da284d-32e4-4afc-a599-7853946fe8c9" />


**Commented out**:
<img width="1440" alt="Screenshot 2025-01-25 at 4 52 55 PM" src="https://github.com/user-attachments/assets/89e43977-8e59-485a-b1d3-6c61f4d8f558" />

kevwu27 reviewed (changes requested):
Great work you guys! Not sure if this is just on my end, so if the other reviewers could confirm as well that'd be great, but the fonts for the smaller subtext seems to be a different color. 
<img width="331" alt="Screenshot 2025-01-27 at 12 46 54 AM" src="https://github.com/user-attachments/assets/24ba3340-062c-4957-8437-8b6a46f59af0" />

<img width="474" alt="Screenshot 2025-01-27 at 12 46 46 AM" src="https://github.com/user-attachments/assets/4af188ab-a88a-430a-bab1-f0aadeaa99f5" />

Also I noticed that in your components for this page, you included the svg _within_ the file. I think it might be best to have all svg's in the public folder just so everything is organized. The "Back to Sign-In" code is the exact same in both components, so you might even be able to extract it out into its own component to avoid redundancy?

Finally, after fixing the merge conflicts, you guys should be good! Great work once again!

aaronchan32 reviewed (changes requested):
Nice work guys. The bug Kevin mentioned for the color of the text is only present on Mac and not Windows. Make sure to explicitly specify the color of the subtext to be #1C3A29. The arrow next to the "Back to Sign-in" should also be clickable so it can be included inside the anchor tag. Finally, the line break tag `<br />` shouldn't be used as styling/spacing unless it is part of the text. Instead, I would look at code in the `main` branch and see how padding, margins, or gaps are used for spacing!

Good work again!

OrigamiStarz commented:
Thanks for your feedback! Here is what has been fixed:
- Specified instruction text to specifically be `#1C3A29`
- Moved svg into public folder
- Created BackToSignIn component
- Wrapped BackToSignIn component around <a> tag, so clicking the arrow should work now
- Changed spacing from <br /> to CSS margins

Please feel free to let me and Shashwat know if there's anything else we should change.

kevwu27 reviewed (commented):
Don't forget to fix the merge conflicts also!

eshaan-s18 reviewed (approved):
LGTM! Thanks for your hard work guys. I'm approving this and merging it
